### PR TITLE
UAF-4155 Non-Check Non-ACH Payments Step 11 (Non-Taxable Vendor Payme…

### PR DESCRIPTION
**Pull request by mmoen**
…nts)

Modified VendorParameterConstants.java, PurapKeyConstants.java, and PurapPropertyConstants.java to add constants for use in other files.
Added new file PurapService.java that adds method isItemTypeConflictWithTaxPolicy used in validation.
Added new file PurapServiceImpl.java that implements the method in PurapService.
Added new file ItemTypeTaxPolicyValidation.java to validate Requisition item type against account tax policy.
Modified PurapValidatorDefinitions.xml to point to the new ItemTypeTaxPolicyValidation.
Modified RequisitionValidation.xml, added itemTypeTaxPolicyValidation to the list of validations.
Modified spring-purap.xml to point to the new PurapServiceImpl.

### CODE REVIEW

File: kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/service/PurapService.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+public interface PurapService extends org.kuali.kfs.module.purap.document.service.PurapService {
+    
+    /**
+     * determine whether the item type is conflicted with tax policy
+     * 
+     * @param purchasingAccountsPayableDocument the given purap document
+     * @param item the given item
+     * @return true if the item type is conflicted with tax policy
+     */
+    boolean isItemTypeConflictWithTaxPolicy(PurchasingAccountsPayableDocument purchasingAccountsPayableDocument, PurApItem item);
```
Comment (ghost) [inline kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/service/PurapService.java]: Should be:
public boolean isItemTypeConflictWithTaxPolicy(PurchasingAccountsPayableDocument purchasingAccountsPayableDocument, PurApItem item);

File: kfs-purap/src/main/resources/edu/arizona/kfs/module/purap/document/validation/configuration/RequisitionValidation.xml [hunk 2]
```diff
@@ -50,5 +50,14 @@
       </list>
     </property>
   </bean>
+
```
Comment (ghost) [inline kfs-purap/src/main/resources/edu/arizona/kfs/module/purap/document/validation/configuration/RequisitionValidation.xml]: The org.kuali version of RequisitionValidation.xml already has the PurchasingAccountsPayable-itemTypeTaxPolicyValidation bean defined to the route document event (see line 80).

If you had to provide this override in order to make this Requistion validation work, I suggest flipping lines 30 and 31 in edu.arizona.kfs.module.purap.spring-purap.xml. PurapValidatorDefinitions.xml should be first in the list of validators.

**Review by ghost — COMMENTED** [review]
Initial review complete. Please see my comments.

File: kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/service/PurapService.java (hunk 1 above)
Comment (mmoen) [reply kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/service/PurapService.java]: Fixed

File: kfs-purap/src/main/resources/edu/arizona/kfs/module/purap/document/validation/configuration/RequisitionValidation.xml (hunk 2 above)
Comment (mmoen) [reply kfs-purap/src/main/resources/edu/arizona/kfs/module/purap/document/validation/configuration/RequisitionValidation.xml]: You are correct. I tested without this code and everything worked as expected.  This has been reverted.

**Comment by ghost** [comment]
Looks good!

**Pull request merged**